gateway/gateway.go

```diff
@@ -1479,6 +1471,39 @@ func (gateway *HandleT) StartAdminHandler(ctx context.Context) error {
 	return g.Wait()
 }
 
+func (gateway *HandleT) maxActiveClientMiddleware(next http.Handler) http.Handler {
+	atomic.AddInt32(&gateway.activeClientCount, 1)
+	defer func() {
+		atomic.AddInt32(&gateway.activeClientCount, -1)
+	}()
+
+	//this is to make sure that we don't have more than `maxClient` in-memory at any point of time. As, having more http client than `maxClient`
+	// may lead to gateway OOM kill.
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if maxActiveClients != 0 {
+			select {
+			case gateway.activeClient <- struct{}{}:
+				defer func() {
+					<-gateway.activeClient
+				}()
+				next.ServeHTTP(w, r)
+			default:
+				http.Error(w, "server is under pressure", response.GetStatusCode(response.TooManyRequestsInMemory))
+				return
+			}
+		}
+	})
+}
+
+func (gateway *HandleT) statMiddleware(next http.Handler) http.Handler {
```

Comment:
@chandumlg @lvrach  Is there a place where we collect response status code metrics per path? IMO this is the place to capture such metrics so that we don't (unintentionally) miss anything.

So to sum up, I propose that:
1. timer stat should be enhanced with the `code` tag
2. introduce a new counter stat `gateway.response_count` using the same set of tags (`reqType`, & `code`)

_Note:_
With regards to naming, I would normally expect to see the following names being used here:
```
gateway.request_time{path="/v1/identify", code="200" }
gateway.request_count{path="/v1/identify", code="200" }
```

Reply:
Right now, we don't have response status code in the metrics. 

Can we do this in another pr, where we have a task to clean up gateway response codes? CC @fracasula